lib/encumber.rb

```diff
@@ -374,42 +377,74 @@ def move_mouse_to_point(x, y)
+      raise "An error occured: #{result}" if result["result"] !='OK'
+      result = command('simulateMouseUp', id_for_element(xpath), flags, $CPLeftMouseUp)
+      raise "An error occured: #{result}" if result["result"] !='OK'
     end
 
-    def simulate_left_click_on_point(x, y, flags=[])
-      move_mouse_to_point(x, y)
+    def simulate_left_click_on_point(x, y, flags=[], move_mouse=true)
+      move_mouse_to_point(x, y) if move_mouse
 
-      result = command('simulateLeftClickOnPoint', x, y, flags)
+      result = command('simulateMouseDownOnPoint', x, y, flags, $CPLeftMouseDown)
+      raise "An error occured: #{result}" if result["result"] !='OK'
+      result = command('simulateMouseUpOnPoint', x, y, flags, $CPLeftMouseUp)
+      raise "An error occured: #{result}" if result["result"] !='OK'
     end
 
-    def simulate_double_click(xpath, flags=[])
+    def simulate_right_click(xpath, flags=[], move_mouse=true)
       points = points_for_element(xpath)
-      move_mouse_to_point(points[0], points[1])
+      move_mouse_to_point(points[0], points[1]) if move_mouse
 
-      result = command('simulateDoubleClick', id_for_element(xpath), flags)
-      raise "View not found: #{xpath} - #{result}" if result["result"] != 'OK'
+      result = command('simulateMouseDown', id_for_element(xpath), flags, $CPRightMouseDown)
+      raise "An error occured: #{result}" if result["result"] !='OK'
+      result = command('simulateMouseUp', id_for_element(xpath), flags, $CPRightMouseUp)
+      raise "An error occured: #{result}" if result["result"] !='OK'
     end
 
-    def simulate_double_click_on_point(x, y, flags=[])
-      move_mouse_to_point(x, y)
+    def simulate_right_click_on_point(x, y, flags=[], move_mouse=true)
+      move_mouse_to_point(x, y) if move_mouse
 
-      result = command('simulateDoubleClick', x, y, flags)
+      result = command('simulateMouseDownOnPoint', x, y, flags, $CPRightMouseDown)
+      raise "An error occured: #{result}" if result["result"] !='OK'
+      result = command('simulateMouseUpOnPoint', x, y, flags, $CPRightMouseUp)
+      raise "An error occured: #{result}" if result["result"] !='OK'
+    end
+
+    def simulate_double_click(xpath, flags=[])
+      simulate_left_click(xpath, flags)
+      simulate_left_click(xpath, flags, false)
+    end
+
+    def simulate_double_click_on_point(x, y, flags=[])
+      simulate_left_click_on_point(x, y, flags)
+      simulate_left_click_on_point(x, y, flags, false)
     end
 
     def simulate_dragged_click_view_to_view(xpath1, xpath2, flags=[])
       points = points_for_element(xpath1)
       move_mouse_to_point(points[0], points[1])
-
-      result = command('simulateDraggedClickViewToView', id_for_element(xpath1), id_for_element(xpath2), flags)
-      raise "View not found: #{xpath1} or #{xpath2}- #{result}" if result["result"] != 'OK'
+      result = command('simulateMouseDown', id_for_element(xpath1), flags, $CPLeftMouseDown)
```

Comment:
What is the point to have `$CPLeftMouseDown` ? The cappuccino method will know that we want to generate a leftMouseDown